### ISSUE TITLE
Windows build

### ISSUE THREAD

**Issue by BigJoe01** [rethinkdb/rethinkdb#6223]
On windows, 

std::function don't support lambda [this] capture

src : real_reql_cluster_interface.cc , line 51
      real_reql_cluster_interface_t constructor
      m_changefeed_client initialization

vs error code :

 real_reql_cluster_interface.cc(55): error C2664: 'ql::changefeed::client_t::client_t(const ql::changefeed::client_t &)': cannot convert argument 2 from 'real_reql_cluster_interface_t::<lambda_56eb3f4f88c05fc89428f33402c85b51>' to 'const std::function<namespace_interface_access_t (const namespace_id_t &,signal_t *)> &'

Repo 2.3.x windows
Visual studio 2015 update 3

**Comment by AtnNn** [comment]
Thanks for the report. A fix would be welcome. I might try to fix this next time I reboot into Windows.

**Comment by srh** [comment]
I tried to build v2.3.x but it was too difficult for me to get working (compared to the `next` branch, which apparently has more Windows work put into it -- it's a lot easier to get working).

I don't think the problem is with [this] capture per se -- it's used in a lot of other places and converts to a std::function just fine in the `next` branch, on Windows.

**Comment by AtnNn** [comment]
There seems to be a separate `v2.3.x-windows` branch. @BigJoe01 Is that the branch you are using?

**Comment by AtnNn** [comment]
I was able to reproduce this error. Working on a fix now.

**Comment by AtnNn** [comment]
Here's what it boils down to:

```
#include <functional>

class namespace_interface_access_t;

class client_t  {
	std::function<
		namespace_interface_access_t()
	> const namespace_source;
};

class namespace_interface_access_t { };

namespace_interface_access_t(*f1)();

auto f2 = std::function<namespace_interface_access_t()>(f1);
```

Moving the definition of `namespace_interface_access_t` to the top allows this code to build. The error does not seem to be caused by lambdas or capturing `this`.

**Comment by BigJoe01** [comment]
Hi,
The problem is this scope.  std::function not support scoped function.

**Comment by AtnNn** [comment]
I suspect that in this case, the error is caused by a broken instantiation of `std::function<foo()>` when `foo` has not yet been defined.

Fixed in #6271

**Comment by AtnNn** [comment]
A fix (#6271) has been merged into the `v2.3.x-windows` branch
